File: app/models/scraper.rb
class Scraper
  attr_accessor :url

  def initialize(url)
    @url = url
  end

  def get_hrefs
    response = HTTParty.get(@url)
    doc = Nokogiri::HTML(response)
    hrefs = doc.css('a').map {|link| link['href']}
    hrefs.map do |href|
      if href.index('http')
        href
      else
        if @url.last == '/'
          @url + href
        else
          @url + '/' + href
        end
      end
    end

  end

  def get_images
    response = HTTParty.get(@url)
    doc = Nokogiri::HTML(response)
    images = doc.css('img').map {|image| image['src']}
    images.map do |image|
      if image.index('http')
        image
      else
        if @url.last == '/'
          @url + image
        else
          @url + '/' + image
        end
      end
    end

  end

end

File: app/assets/javascripts/scraper.js
$(function() {

  function getLinks(url) {
    return $.ajax({
      url: '/scraper/get_links',
      type: 'post',
      data: { url: url }
    });
  }

  function getImages(url) {
    return $.ajax({
      url: '/scraper/get_images',
      type: 'post',
      data: { url: url }
    });
  }

  $('#links').on('click', function() {
    var url = $('#url').val();

    getLinks(url).done(function(data) {
      console.log(data);

      for (var i=0; i<data.length; i++) {
        var $pTag = $('<p>'),
            $link = $('<a>');
        $link.attr('href', data[i]);
        $link.text(data[i]);
        $pTag.append($link);
        $('body').append($pTag);
      }
    });
  });

  $('#images').on('click', function() {
    var url = $('#url').val();

    getImages(url).done(function(data) {
      console.log(data);

      for (var i=0; i<data.length; i++) {
        var $pTag = $('<p>'),
            $image = $('<img>');
        $image.attr('src', data[i]);
        $pTag.append($image);
        $('body').append($pTag);
      }
    });
  });

});
